Replicator/Replicator.cc

```diff
@@ -1297,16 +1306,24 @@ namespace litecore::repl {
             }
         }
 
+        WorkerHandler handler = _workerHandlers.useLocked<WorkerHandler>([&](WorkerHandlers& handlers) {
```

Comment:
It's rather heavy here to lock for every message coming. I suppose this method itself is synchronized. (I may be wrong) If so. the only places I see that require synchronization is  .clear when terminate or stop. If so, we may avoid clear simply by not calling handlers. Right?

Reply:
This method is not synchronized. That's why the comment at the top says `// This method is NOT called on the Replicator's queue/thread! It must be thread-safe.` Anyway, a mutex is super lightweight compared to receiving an RPC message.

Plus, this is a big improvement on the previous implementation where _all_ incoming messages got queued through the Replicator object's actor queue and couldn't not be handled concurrently.